3-enrich/spark-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich.spark/EnrichJobConfig.scala

```diff
@@ -0,0 +1,147 @@
+  override val igluConfig: String = "",
+  override val local: Boolean = false,
+  etlTstamp: Long = 0L
+) extends EnrichJobConfig
+
+/**
+ * Case class representing the configuration for the enrich job.
+ * @param inFolder Folder where the input events are located
+ * @param inFormat Collector format in which the data is coming in
+ * @param outFolder Output folder where the enriched events will be stored
+ * @param badFolder Output folder where the malformed events will be stored
+ * @param enrichments JSON representing the enrichments that need performing
+ * @param igluConfig JSON representing the Iglu configuration
+ * @param local Whether to build a registry from local data
+ * @param etlTstamp Timestamp at which the job was launched
+ */
+case class ParsedEnrichJobConfig(
+  override val inFolder: String,
+  override val inFormat: String,
+  override val outFolder: String,
+  override val badFolder: String,
+  override val enrichments: String,
+  override val igluConfig: String,
+  override val local: Boolean,
+  etlTstamp: DateTime,
+  filesToCache: List[(URI, String)]
+) extends EnrichJobConfig
+
+object EnrichJobConfig {
+  private val parser = new scopt.OptionParser[RawEnrichJobConfig]("EnrichJob") {
+    head("EnrichJob")
+    opt[String]("input-folder").required().valueName("<input folder>")
+      .action((f, c) => c.copy(inFolder = f))
+      .text("Folder where the input events are located")
+    opt[String]("input-format").required().valueName("<input format>")
+      .action((f, c) => c.copy(inFormat = f))
+      .text("The format in which the collector is saving data")
+    opt[String]("output-folder").required().valueName("<output folder>")
+      .action((f, c) => c.copy(outFolder = f))
+      .text("Output folder where the enriched events will be stored")
+    opt[String]("bad-folder").required().valueName("<bad folder>")
+      .action((f, c) => c.copy(badFolder = f))
+      .text("Output folder where the malformed events will be stored")
+    opt[String]("enrichments").required().valueName("<enrichments>")
+      .action((e, c) => c.copy(enrichments = e))
+      .text("Directory where the JSONs describing the enrichments are stored")
+    opt[String]("iglu-config").required().valueName("<iglu config>")
+      .action((i, c) => c.copy(igluConfig = i))
+      .text("Iglu resolver configuration")
+    opt[Long]("etl-timestamp").required().valueName("<ETL timestamp>")
+      .action((t, c) => c.copy(etlTstamp = t))
+      .text("Timestamp at which the job was launched, in milliseconds")
+    opt[Unit]("local").hidden()
+      .action((_, c) => c.copy(local = true))
+      .text("Whether to build a local enrichment registry")
+    help("help").text("Prints this usage text")
+  }
+
+  /**
+   * Load a EnrichJobConfig from command line arguments.
+   * @param args The command line arguments
+   * @return The job config or one or more error messages boxed in a Scalaz ValidationNel
+   */
+  def loadConfigFrom(
```

Comment:
I don't like having the return type being the first thing one the line, just a matter of personal preference.